classes/profile_helper.php

```diff
@@ -145,9 +145,19 @@ public static function get_min_duration_for_reason(int $reason, bool $usecache =
                      WHERE $reasons != ?
                   ORDER BY duration DESC
                      ";
-            $resultset = $db->get_records_sql($sql, [
-                profile::REASON_NONE,
-            ], $quota - 1, 1); // Will fetch the Nth item based on the quota.
+
+            try {
+                $resultset = $db->get_records_sql(
+                    $sql,
+                    [profile::REASON_NONE],
+                    $quota - 1, // Will fetch the Nth item based on the quota.
+                    1
+                );
+            } catch (\dml_exception $e) {
+                debugging('tool_excimer: Failed to get min duration: '.$e->getMessage());
+                return null;
```

Comment:
the calling code expects a number and hasn't been touched. Thinking that we could just return a very large number like 1 week so everything is smaller than it

Reply:
I checked the calling code and it all looks compatible. Returning a large number could also work. See my alternate fix which could be a better way of solving the problem.